public/app/features/dashboard/components/ShareModal/ShareLink.tsx

```diff
@@ -44,23 +47,33 @@ export class ShareLink extends PureComponent<Props, State> {
   }
 
   componentDidUpdate(prevProps: Props, prevState: State) {
-    const { useCurrentTimeRange, includeTemplateVars, selectedTheme } = this.state;
+    const { useCurrentTimeRange, includeTemplateVars, useShortUrl, selectedTheme } = this.state;
     if (
       prevState.useCurrentTimeRange !== useCurrentTimeRange ||
       prevState.includeTemplateVars !== includeTemplateVars ||
-      prevState.selectedTheme.value !== selectedTheme.value
+      prevState.selectedTheme.value !== selectedTheme.value ||
+      prevState.useShortUrl !== useShortUrl
     ) {
       this.buildUrl();
     }
   }
 
   buildUrl = () => {
     const { panel } = this.props;
-    const { useCurrentTimeRange, includeTemplateVars, selectedTheme } = this.state;
+    const { useCurrentTimeRange, includeTemplateVars, useShortUrl, selectedTheme } = this.state;
 
     const shareUrl = buildShareUrl(useCurrentTimeRange, includeTemplateVars, selectedTheme.value, panel);
     const imageUrl = buildImageUrl(useCurrentTimeRange, includeTemplateVars, selectedTheme.value, panel);
-    this.setState({ shareUrl, imageUrl });
+
+    if (useShortUrl) {
+      getBackendSrv()
+        .post(`/api/short-urls`, {
+          path: shareUrl.replace(buildHostUrl(), ''),
```

Comment:
Hadn't look closely at the frontend code earlier. I would like API to take the full URL (`window.location`) so that the backend can hold all logic for validating, parsing URL and what parts to use for storing. Then when reading the url from the database the backend know how it's been saved and can redirect it properly.

Reply:
Just to confirm - that I understand it correctly: 
- we should send full url from frontend to backend
- at backend, before we store url in database, we want to remove host/port/appSubUrl and save only path as it is saved in the database now, right?

Reply:
1. Yes
1. Yes
1. If you mean when saving I suppose that needs to change yes

Let us know if you need help with these changes

Reply:
That should be pretty straight forward, just updating `path: shareUrl.replace(buildHostUrl(), '')` to `path: shareUrl` and in createShortURL, change `uid, err := service.CreateShortURL(c.SignedInUser, strings.TrimPrefix(cmd.Path, "/"))` to `uid, err := service.CreateShortURL(c.SignedInUser, strings.TrimPrefix(cmd.Path, setting.AppUrl))`. @aknuds1 I have accidentally pushed the changes to https://github.com/grafana/grafana/tree/arve/mistersquishy-shorturls, but I can see that you are using different branch now. So I'll probably wait until you commit the changes here.

Reply:
Yes sounds/looks good. But instead of doing `buildShortUrl` client side I would delegate that responsibility to the backend. So when creating a new short URL, return the full URL instead of UID.